Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls.cpp

```diff
@@ -393,21 +393,43 @@ struct StackFramePlusRet {
   uint64_t Pad;
 };
 
+#ifdef _M_ARM_64
+#define ExitWithoutStack(Result) \
+  __asm volatile( \
+    "mov x8, %[SyscallNum];" \
+    "mov x0, %[Result];" \
+    "svc #0;" \
+    :: [SyscallNum] "i" (SYSCALL_DEF(exit)) \
+    , [Result] "r" (Result) \
+    : "memory", "x0", "x8");
+#else
+#define ExitWithoutStack(Result) \
+  __asm volatile( \
+    "mov %[Result], %%rdi;" \
+    "syscall;" \
+    :: "a" (SYSCALL_DEF(exit)) \
+    , [Result] "r" (Result) \
+    : "memory", "rdi");
```

Comment:
These can be `[[noreturn]]` functions instead, can't they?

Reply:
They cannot since debug builds will try to use the stack and we can't guarantee tail-call optimization.